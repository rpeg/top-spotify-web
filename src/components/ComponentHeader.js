import React from 'react';
import { Row } from 'react-bootstrap';
import PropTypes from 'prop-types';

const ComponentHeader = ({ title }) => (
  <Row style={{ padding: '2em 0 0 0' }}>
    <p
      className="text-m"
      style={{
        textAlign: 'left', backgroundColor: '#FFFFFF', color: '#191414', padding: '0 .25em 0 .25em',
      }}
    >
      {title}
    </p>
  </Row>
);

ComponentHeader.propTypes = {
  title: PropTypes.string.isRequired,
};

export default ComponentHeader;
