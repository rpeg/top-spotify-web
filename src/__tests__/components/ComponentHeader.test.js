import React from 'react';
import { shallow } from 'enzyme';

import ComponentHeader from '../../components/ComponentHeader';

describe('ComponentHeader', () => {
  const title = 'Test';

  it('should not log errors in console', () => {
    const wrapper = shallow(<ComponentHeader title={title} />);
    const spy = jest.spyOn(global.console, 'error');
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });

  it('renders correctly', () => {
    const wrapper = shallow(<ComponentHeader title={title} />);
    expect(wrapper).toMatchSnapshot();
  });
});
