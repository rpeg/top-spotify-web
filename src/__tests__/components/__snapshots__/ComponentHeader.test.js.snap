// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentHeader renders correctly 1`] = `
<ForwardRef
  noGutters={false}
  style={
    Object {
      "padding": "2em 0 0 0",
    }
  }
>
  <div>
    <p
      className="text-l"
      style={
        Object {
          "fontWeight": "bold",
          "marginBottom": "0",
          "paddingLeft": ".1em",
          "textAlign": "left",
        }
      }
    >
      Test
    </p>
  </div>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <hr
      style={
        Object {
          "borderTop": "1px solid #fff",
          "margin": "0 0 2em 0",
        }
      }
    />
  </div>
</ForwardRef>
`;
